mlxtend/regressor/linear_regression.py

```diff
@@ -69,6 +74,7 @@ def __init__(self, eta=0.01, epochs=50,
         self.random_seed = random_seed
         self.print_progress = print_progress
         self._is_fitted = False
+        self.method = method
 
```

Comment:
Another check to add would be

```python
supported_methods = {'sgd', 'direct', 'svd', 'qr')
if method not in supported_methods:
    raise ValueError('`method` must be in %s. Got %s.' % (
                     supported_methods, method))

Reply:
Should I add these checks in __init__ method...

Reply:
Should I change LinearRegression.ipynb too?

Reply:
> Should I add these checks in __init__ method...

Good point. Yeah, this would be best in the `__init__` method.

> Should I change user_guide LinearRegression.ipynb too?

You could add an example if you like. Otherwise, only the api docs need to be regenerated, because those would be loaded in the Jupyter Notebook (the last cell), which then later gets turned into the HTML documention upon a new version release. In order to do that, simply 

- `cd` into the `mlxtend/docs` directory
- inside the `mlxtend/docs` directory, execute `python3 make_api.py`
- then, rerun ("run all cells") notebook and save it

Reply:
For user_guide I am thinking of quoting from The Elements of Statistical Learning Chapter 3 Lin.Reg. for both methods using QR and SVD...

Reply:
> * `cd` into the `mlxtend/docs` directory
> * inside the `mlxtend/docs` directory, execute `python3 make_api.py`
> * then, rerun ("run all cells") notebook and save it

Okay, doing this too... What about a little theory and formulas behind like they are present for SGD method and analytical method...

Reply:
If you want to contribute some background info on these method (and quoting ESL as a ref), that would be awesome

Reply:
Okay, will make changes

Reply:
I have added info on both QR and SVD in LinearRegression.ipynb. Can you check it out?
Is what I have added sufficient, specially the definition of both of them...